mlflow/data/numpy_dataset.py

```diff
@@ -0,0 +1,140 @@
+import hashlib
+import json
+from typing import List, Optional, Any, Dict, Union
+
+import numpy as np
+import pandas as pd
+
+from mlflow.data.dataset import Dataset
+from mlflow.data.filesystem_dataset_source import FileSystemDatasetSource
+from mlflow.types import Schema
+from mlflow.types.utils import _infer_schema
+
+
+class NumpyDataset(Dataset):
```

Comment:
Can we also introduce a PyFuncConvertibleDatasetMixin mixin that defines a to_pyfunc() method returning a PyFuncInput for use with pyfunc predict? Can we then have NumpyDataset and PandasDataset inherit from PyFuncConvertableDatasetMixin as well as Dataset and add test coverage for numpydataset.to_pyfunc() and pandasdataset.to_pyfunc()?

Implementing to_pyfunc() for these datasets should hopefully be very straightforward because pandas dataframes and numpy arrays are already included in PyFuncInput. For more info, see plugin system design doc :)

Reply:
The motivation here is so users can pass these datasets directly to mlflow.evaluate()